db/change_cache.go

```diff
@@ -208,13 +213,30 @@ func (c *changeCache) Start(initialSequence uint64) error {
 	return nil
 }
 
+// waitForBGTCompletion waits for all the background tasks to finish.
+func waitForBGTCompletion(ch ...chan struct{}) {
+	for _, v := range ch {
+	waitForCompletion:
+		for {
+			select {
+			case <-v:
+				break waitForCompletion
+			default:
```

Comment:
Having a default case here will cause the for loop to spin (without any delay) if v isn't closed yet.

Reply:
The default clause has been removed with the latest changes.